Cmake: Doesn't configure if Scafacos is installed

CMake Error at src/core/electrostatics/CMakeLists.txt:4 (target_link_libraries):
  INTERFACE library can only be used with the INTERFACE keyword of
  target_link_libraries


CMake Error at src/core/electrostatics/scafacos/CMakeLists.txt:2 (add_dependencies):
  Cannot add target-level dependencies to non-existent target "Scafacos".

  The add_dependencies works for top-level logical targets created by the
  add_executable, add_library, or add_custom_target commands.  If you want to
  add file-level dependencies see the DEPENDS option of the add_custom_target
  and add_custom_command commands.

On an Ubuntu system, Scafacos installed system wide using make install

--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -33,7 +33,7 @@ add_dependencies(EspressoConfig myconfig)
 
 add_library(EspressoCore SHARED ${EspressoCore_SRC} config-features.hpp config-features.cpp config-version.cpp)
 
-target_link_libraries(EspressoCore ${LIBRARIES} Actor ObjectInFluid Electrostatics ImmersedBoundary)
+target_link_libraries(EspressoCore ${LIBRARIES} Actor ObjectInFluid ImmersedBoundary Scafacos)
 set_target_properties(EspressoCore PROPERTIES MACOSX_RPATH TRUE)
 
 if(CUDA)

--- a/src/core/electrostatics/CMakeLists.txt
+++ b/src/core/electrostatics/CMakeLists.txt
@@ -1,7 +1,3 @@
-add_library(Electrostatics INTERFACE)
-
 if(SCAFACOS)
-  target_link_libraries(Electrostatics Scafacos)
-
   add_subdirectory(scafacos)
 endif(SCAFACOS)

--- a/src/core/electrostatics/scafacos/CMakeLists.txt
+++ b/src/core/electrostatics/scafacos/CMakeLists.txt
@@ -1,7 +1,7 @@
 include_directories(${SCAFACOS_INCLUDE_DIRS})
-add_dependencies(Scafacos EspressoConfig)
 
 file(GLOB Scafacos_SRC *.cpp)
 add_library(Scafacos SHARED ${Scafacos_SRC})
+add_dependencies(Scafacos EspressoConfig)
 
 target_link_libraries(Scafacos ${SCAFACOS_LDFLAGS})

